### DISCUSSION TITLE
Initial density of MATL layer

### DISCUSSION THREAD

**Discussion by r-broek** [firemodels/fds#13581] [Q&A] [answered]
Dear developers,

I'd like to confirm if ML%RHO_S in e.g. wall.f90 represents the initial density of the MATL layer (user input), given as rho_s(0) in the FDS user's guide 9.2.2.

**Comment by drjfloyd** [comment]
rho_s(0) is not used in pyrolysis in the current release of FDS. Neither Eq. 9.1 nor Eq 9.2 contain rho_s(0). The note at the bottom of the page for those equations that has rho_s(0) is instructions on how to convert the A input for older versions of FDS where rho_s(0) was in the pyrolysis equaiton.

ML%RHO_S is not rho_s(0).  ML%RHO_S is the DENSITY input on MATL.  It will be rho_s(0) when the MATL has a MASS_FRACTION of 1, otherwise it will be a lower value.

**Reply by r-broek** [reply]
Yes I am aware rho_s(0) is not currently used for pyrolysis reactions, but I figured that perhaps the variable still existed within the source code. I wanted to use it for converting A values automatically. Thank you for explaining what ML%RHO_S represents.

**Reply by drjfloyd** [reply, marked as answer]
Look at read_surf in read.f90.   The block of code at line 8535 sets up the intial density and material mass fractions for each layer in a surf.

**Reply by r-broek** [reply]
I will take a look. Thanks again!
